pkg/component/registrycaches/registry_caches.go

```diff
@@ -292,6 +294,22 @@ func (r *registryCaches) computeResourcesDataForRegistryCache(ctx context.Contex
 							Type: corev1.SeccompProfileTypeRuntimeDefault,
 						},
 					},
+					InitContainers: []corev1.Container{
+						{
+							// Mitigation for  https://github.com/distribution/distribution/issues/4478.
+							Name:            "cleanup-volume",
+							Image:           r.values.InitImage,
```

Comment:
Btw, when thinking about it again - can we use the existing registry image? It has sh already. And it will be anyways pulled for the main container? In this way we can avoid maintaining another image. 

This trick is being used for the kube-proxy init container as well - see https://github.com/gardener/gardener/blob/c940694d83ea9e54a4083d6fe9252eefd8c9f55b/pkg/component/kubernetes/proxy/resources.go#L234.